### PR TITLE
Stax Port

### PR DESCRIPTION
**Pull request by chris124567**
Port UI flow to new NBGL library, add UI tests using [ragger](https://github.com/ledgerhQ/ragger), and add go.mod stuff for sialedger.go as go now requires modules.

### CODE REVIEW

**Comment by chris124567** [comment]
It seems like a lot of these issues are either minor or were preexisting (typos, we don't have doxygen documentation, etc).  I'll fix them anyways because it makes the review process go faster :+1:

**Comment by chris124567** [comment]
OK I think all the things that could be addressed from the CI failures have been addressed.  We don't have doxygen documentation, the "typos" that remain are all false positives due to the `flagg` library, we don't have unit tests, the static analyzer errors are just because the txn_state_t struct uses `uint_t` instead of `char` types to be exact, and I don't think we ever supported the NanoSP (?).

File: Makefile [hunk 1]
```diff
@@ -35,10 +35,12 @@ ICONNAME=stax_app_sia.gif
 endif
 
 # The --path argument here restricts which BIP32 paths the app is allowed to derive.
-APP_LOAD_PARAMS = --path "44'/93'" --curve ed25519 $(COMMON_LOAD_PARAMS)
+APP_LOAD_PARAMS = --path "44'/93'" --curve secp256k1 --curve ed25519 $(COMMON_LOAD_PARAMS)
```
Comment (lukechampine) [inline Makefile]: why do we need secp256k1?

File: src/main.c [hunk 2]
```diff
@@ -277,6 +311,7 @@ static handler_fn_t* lookupHandler(uint8_t ins) {
 // and sent in the next io_exchange call.
 static void sia_main(void) {
 	// Mark the transaction context as uninitialized.
+	explicit_bzero(&global, sizeof(global));
```
Comment (lukechampine) [inline src/main.c]: IIRC the C standard requires that these variables be zero-initialized, but whatev

File: src/main.c [hunk 3]
```diff
@@ -371,73 +406,77 @@ static void sia_main(void) {
...
+                !(U4BE(G_io_seproxyhal_spi_buffer, 3) &      //
+                  SEPROXYHAL_TAG_STATUS_EVENT_FLAG_USB_POWERED)) {
+                THROW(EXCEPTION_IO_RESET);
+            }
+            /* fallthrough */
+        case SEPROXYHAL_TAG_DISPLAY_PROCESSED_EVENT:
+#ifdef HAVE_BAGL
+            UX_DISPLAYED_EVENT({});
+#endif  // HAVE_BAGL
+#ifdef HAVE_NBGL
```
Comment (lukechampine) [inline src/main.c:419]: I thought this was redundant, but I see now that it's `NBGL`, not `BAGL`. Nice of them to make them look so similar...

all of these `ifdef`s bum me out, but not much we can do about it I guess

File: src/main.c [hunk 4]
```diff
@@ -371,73 +406,77 @@ static void sia_main(void) {
...
-	case SEPROXYHAL_TAG_TICKER_EVENT:
-		UX_TICKER_EVENT(G_io_seproxyhal_spi_buffer, {});
-		break;
-
-	default:
-		UX_DEFAULT_EVENT();
-		break;
-	}
+uint8_t io_event(uint8_t channel) {
+    (void)channel;
```
Comment (lukechampine) [inline src/main.c]: should be `UNUSED`?

File: Makefile (hunk 1 above)
Comment (chris124567) [reply Makefile]: It didn't make sense to me but some Ledger tool was warning about it.  Apparently https://github.com/LedgerHQ/ledger-app-database/blob/main/app-load-params-db.json#L1194 is wrong.  I guess we could make a PR there?

Comment (chris124567) [reply Makefile]: https://github.com/LedgerHQ/ledger-app-database/pull/27

**Comment by chris124567** [comment]
If there's no other problems I guess this can be merged?  Unless we want to have Ledger review it in this PR.  The only remaining (minor) issue is the app database load parameters but we can't control that here anyways.

**Comment by chris124567** [comment]
Please also consider https://github.com/SiaFoundation/app-sia-x/commit/9975369ee02fc6aaab7d37a4964a1e2067026b29 in this same stax-port branch but which for some reason doesn't show up on this page, which fixes the static analyzer warnings and adds Stax USB support to sialedger.go.

**Pull request merged**